sirepo/srunit.py

```diff
@@ -782,6 +789,32 @@ def change_to_redirect(self, uri):
     def header_get(self, name):
         return self._headers[name]
 
+    def _convert_http_status(self, sr_args):
+        """set self.status_code
+
+        Returns:
+            bool: True if is a redirect
+        """
+        from pykern import pkdebug
+        from sirepo import uri
+
+        if sr_args.routeName == "httpException":
+            self.status_code = sr_args.params.code
+            return False
+        elif sr_args.routeName == "httpRedirect":
+            self.change_to_redirect(sr_args.params.uri)
+            return True
+        elif not uri.is_sr_exception_only(
+            self._test_client.sr_sim_type, sr_args.routeName
+        ):
+            self.change_to_redirect(
+                uri.local_route(
+                    self._test_client.sr_sim_type, sr_args.routeName, sr_args.params
+                )
+            )
+            return True
+        return False
```

Comment:
what is this case? Why doesn't self.status_code get set?

Reply:
[This change to reply](https://github.com/radiasoft/sirepo/pull/7016/files/88bdeb8757387e425cba3e77947995fe7230b260#diff-ced4e1c88fa61df60076e5700da33a2ce965f4be4ba149a74d4c8bb4f3d28b66L787) is why I modularized the code. srException is used for redirects and for `:srExceptionOnly` cases. That causes problems in the non-websocket case with get, which doesn't return the exception but simply redirects. We can only return json on a post.

Reply:
I'm not entirely following. My question is, in what case will we reach line 816? This function's documented effect is "set self.status_code". If we reach line 816, that doesn't happen. An explanation should be documented here.

Reply:
Currently, the only case is sbatchLogin. There are three routes with :srExceptionOnly: sbatchLogin, httpException, and httpRedirect. The first two cases are special, and handled above. sirepo.reply checks this and uses all three cases. The sbatchLogin case is [here](https://github.com/radiasoft/sirepo/pull/7016/files/88bdeb8757387e425cba3e77947995fe7230b260#diff-ced4e1c88fa61df60076e5700da33a2ce965f4be4ba149a74d4c8bb4f3d28b66R793).